components/motor/tmcstepper/stepper_motor_tmc_test.go

```diff
@@ -139,17 +125,11 @@ func TestTMCStepperMotor(t *testing.T) {
 	ctx := context.Background()
 	logger := logging.NewTestLogger(t)
 	c := make(chan []byte)
-	b := &fakeboard.Board{
-		Named: board.Named("foo").AsNamed(),
-	}
-	b.GPIOPins = map[string]*fakeboard.GPIOPin{}
-	b.SPIs = map[string]*fakeboard.SPI{}
-	b.SPIs["main"] = &fakeboard.SPI{FIFO: c}
+	fakeSpi := fakeboard.SPI{FIFO: c}
 
-	deps := resource.Dependencies(map[resource.Name]resource.Resource{b.Name(): b})
+	var deps resource.Dependencies
```

Comment:
is it possible to define this once at the top of the file? or does that cause race condition issues (or any issues really)?

Reply:
No race conditions. It's here because that's where it originally was. I kinda want to define it just before its first use, down on line 162, but because it keeps getting used later, whoever wrote this might be right to define it at the top of the function, with the logger and fake SPI bus.

Defining it as a global variable at the top of the file seems brittle: we can't make it `const` because it has a nontrivial type, and if we make it mutable and someone changes it (e.g., because they're testing the enable pin), those side effects will bleed over from one test to the next, causing confusion if a test ever fails.

Reply:
yeah that makes sense, it's fine here then